app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/SignatureVerifier.java

```diff
@@ -0,0 +1,78 @@
+package com.appsmith.server.helpers;
+
+import com.appsmith.server.exceptions.AppsmithError;
+import com.appsmith.server.exceptions.AppsmithException;
+import org.apache.commons.collections.CollectionUtils;
+import org.bouncycastle.crypto.signers.Ed25519Signer;
+import org.bouncycastle.crypto.params.AsymmetricKeyParameter;
+import org.bouncycastle.crypto.util.OpenSSHPublicKeyUtil;
+import org.pf4j.util.StringUtils;
+import org.springframework.http.HttpHeaders;
+
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Base64;
+import java.util.Objects;
+
+import static com.appsmith.server.constants.ApiConstants.APPSMITH_SIGNATURE;
+import static com.appsmith.server.constants.ApiConstants.DATE;
+
+public class SignatureVerifier {
+    private static final String publicVerificationKey = "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAICNwJ+zx2opXjjOga/YyzRxb2czvNgQ/twA+miCKDIX3 appsmith";
```

Comment:
Can we rename this constant to `PUBLIC_VERIFICATION_KEY` please?

Reply:
Also, looks like we only need the key part of this string. So, how about:

```suggestion
    // Public key for verifying signatures with ED25519 scheme.
    private static final String PUBLIC_VERIFICATION_KEY = "AAAAC3NzaC1lZDI1NTE5AAAAICNwJ+zx2opXjjOga/YyzRxb2czvNgQ/twA+miCKDIX3";
```

Then we don't have to do the `.split` to get the key every time.